homeassistant/components/emoncms/sensor.py

```diff
@@ -195,7 +194,27 @@ def __init__(
         elif unit_of_measurement == "hPa":
             self._attr_device_class = SensorDeviceClass.PRESSURE
             self._attr_state_class = SensorStateClass.MEASUREMENT
+        self._update_attributes(elem)
+        self._update_value(elem)
+
+    def _update_attributes(self, elem: dict[str, Any]) -> None:
+        """Update extra attributes."""
+        self._attr_extra_state_attributes = {
+            ATTR_FEEDID: elem["id"],
+            ATTR_TAG: elem["tag"],
+            ATTR_FEEDNAME: elem["name"],
+        }
+        if elem["value"] is not None:
+            self._attr_extra_state_attributes[ATTR_SIZE] = elem["size"]
+            self._attr_extra_state_attributes[ATTR_USERID] = elem["userid"]
+            self._attr_extra_state_attributes[ATTR_LASTUPDATETIME] = elem["time"]
+            self._attr_extra_state_attributes[ATTR_LASTUPDATETIMESTR] = (
+                template.timestamp_local(float(elem["time"]))
+            )
 
+    def _update_value(self, elem: dict[str, Any]) -> None:
+        """Update sensor value."""
```

Comment:
```suggestion
```

Reply:
OK thanks :-) it's better to keep things simple